src/hotspot/share/classfile/stackMapTableFormat.hpp

```diff
@@ -549,7 +549,7 @@ class append_frame : public stack_map_frame {
 
   static u1 appends_to_frame_type(int appends) {
     assert(appends > 0 && appends < 4, "Invalid append amount");
-    return 251 + appends;
+    return static_cast<u1>(251 + appends);
```

Comment:
The problem I have with assert + unchecked cast in general is that future maintainers might accidentally remove or break the assert check so we are left with an unchecked cast that truncates out-of-range values.  For that reason I would like to see a checked_cast here.  Even if we could do something like `RangeCheckedInt<1,3> appends` and get rid of the assert, I would probably keep the checked_cast because it catches human error.

Reply:
Ok, I'll change both of them in this file.